diff --git a/karpenter-provider-flex/vendor/github.com/Azure/karpenter-provider-azure/pkg/controllers/nodeclaim/inplaceupdate/controller.go b/karpenter-provider-flex/vendor/github.com/Azure/karpenter-provider-azure/pkg/controllers/nodeclaim/inplaceupdate/controller.go
index fb46b1a..08cd5e4 100644
--- a/karpenter-provider-flex/vendor/github.com/Azure/karpenter-provider-azure/pkg/controllers/nodeclaim/inplaceupdate/controller.go
+++ b/karpenter-provider-flex/vendor/github.com/Azure/karpenter-provider-azure/pkg/controllers/nodeclaim/inplaceupdate/controller.go
@@ -252,6 +252,7 @@ func (c *Controller) Register(_ context.Context, m manager.Manager) error {
 		For(
 			&karpv1.NodeClaim{},
 			builder.WithPredicates(
+				nodeclaimutils.UsingAKSNodeClassPredicate(),
 				predicate.Or(
 					predicate.GenerationChangedPredicate{}, // Note that this will trigger on pod restart for all Machines.
 				),
diff --git a/karpenter-provider-flex/vendor/github.com/Azure/karpenter-provider-azure/pkg/utils/nodeclaim/nodeclaim.go b/karpenter-provider-flex/vendor/github.com/Azure/karpenter-provider-azure/pkg/utils/nodeclaim/nodeclaim.go
index 6b3719b..60b4631 100644
--- a/karpenter-provider-flex/vendor/github.com/Azure/karpenter-provider-azure/pkg/utils/nodeclaim/nodeclaim.go
+++ b/karpenter-provider-flex/vendor/github.com/Azure/karpenter-provider-azure/pkg/utils/nodeclaim/nodeclaim.go
@@ -24,6 +24,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"

 	armcompute "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v7"
@@ -33,6 +34,25 @@ import (
 	"github.com/Azure/karpenter-provider-azure/pkg/utils"
 )

+// UsingAKSNodeClass tells if the giving node claim referencing AKS node class.
+func UsingAKSNodeClass(nodeClaim *karpv1.NodeClaim) bool {
+	if nodeClaim.Spec.NodeClassRef == nil {
+		return false
+	}
+	return nodeClaim.Spec.NodeClassRef.Kind == "AKSNodeClass"
+}
+
+// UsingAKSNodeClassPredicate creates a predicate to filter node claim using AKS node class.
+func UsingAKSNodeClassPredicate() predicate.Funcs {
+	return predicate.NewPredicateFuncs(func(object client.Object) bool {
+		nodeClaim, ok := object.(*karpv1.NodeClaim)
+		if !ok {
+			return false
+		}
+		return UsingAKSNodeClass(nodeClaim)
+	})
+}
+
 // GetAKSNodeClass resolves the AKSNodeClass from the NodeClaim's NodeClassRef.
 // If the NodeClass for the nodeClaim has DeletionTimestamp set, an error is returned.
 func GetAKSNodeClass(ctx context.Context, kubeClient client.Client, nodeClaim *karpv1.NodeClaim) (*v1beta1.AKSNodeClass, error) {
