diff --git a/vendor/github.com/Azure/karpenter-provider-azure/pkg/operator/operator.go b/vendor/github.com/Azure/karpenter-provider-azure/pkg/operator/operator.go
index 397760b..143d58b 100644
--- a/vendor/github.com/Azure/karpenter-provider-azure/pkg/operator/operator.go
+++ b/vendor/github.com/Azure/karpenter-provider-azure/pkg/operator/operator.go
@@ -314,7 +314,13 @@ func getVnetGUID(ctx context.Context, creds azcore.TokenCredential, cfg *auth.Co
 }

 // WaitForCRDs waits for the required CRDs to be available with a timeout
-func WaitForCRDs(ctx context.Context, timeout time.Duration, config *rest.Config, log logr.Logger) error {
+func WaitForCRDs(
+	ctx context.Context,
+	timeout time.Duration,
+	config *rest.Config,
+	log logr.Logger,
+	otherObjs ...runtime.Object,
+) error {
 	gvk := func(obj runtime.Object) schema.GroupVersionKind {
 		return lo.Must(apiutil.GVKForObject(obj, scheme.Scheme))
 	}
@@ -323,6 +329,9 @@ func WaitForCRDs(ctx context.Context, timeout time.Duration, config *rest.Config
 		gvk(&karpv1.NodeClaim{}),
 		gvk(&v1beta1.AKSNodeClass{}),
 	}
+	for _, o := range otherObjs {
+		requiredGVKs = append(requiredGVKs, gvk(o))
+	}

 	client, err := rest.HTTPClientFor(config)
 	if err != nil {
